universe/Building.h

```diff
@@ -214,9 +214,8 @@ class BuildingTypeManager {
 
 private:
     BuildingTypeManager();
-    ~BuildingTypeManager();
 
-    std::map<std::string, BuildingType*> m_building_types;
+    std::map<std::string, std::unique_ptr<BuildingType>> m_building_types;
```

Comment:
> First, it is static, global, parses the building types once and never de-allocates them.
>
> Second, I checked ~52 uses of its returned pointers and they are correctly used and never stored.

I'd rather not build in assumptions that the contents of manager classes will stay valid. Particularly due to the potential for multirhreading in network messaging between the server and clients, and proposals to update the content when switching content directories from default and back. Even within a function's local scope, a raw pointer could be invalidated in another thread. Making the pointer point to a const type seems better to specify that the returned shared pointer shouldn't be used to modify the object.